docs/2.0/docs/patcher/concepts/grouping.md

```diff
@@ -0,0 +1,76 @@
+    :::
+
+## Implementation Discussion
+
+When generating pull requests in CI, the workflow generally involves first invoking `patcher report` to identify what updates are available, then `patcher update` to create PRs applying those updates. Patcher does not have a built-in way to specify the grouping strategy by name, instead the grouping strategies are implemented with flags to the report and update commands.
+
+### Patcher Report
+
+Patcher [report](/2.0/reference/patcher/#report) accepts a `--include-dirs` argument. When passed, this will cause `patcher report` to only output updates matching the glob pattern specified. This allows the developer to in essence limit the resulting PR to only a single environment. By invoking `patcher report` multiple times with different `--include-dirs` arguments, the developer can in effect create different PR worklfows for each environment.
+
+Patcher `report` outputs in JSON which can be inspected or iterated over to achieve desired behaviors.
+
+### Patcher Update
+
+Patcher [update](/2.0/reference/patcher/#update) accepts a `--target` argument. This argument can be specified one or many times, and limits which dependencies patcher will include in the generated pull request. By invoking `patcher update` multiple times with different `--target` arguments, the developer can in effect create different PR workflows for each dependency.
+
+## Implementation Walkthrough
+
+### Full-Consolidation
+To implement a full-consolidation workflow, do not pass either `--include-dirs` or `--target` parameters to patcher `report` and `update`.  The output of patcher `report` will include all updates that are available, and the effect of patcher `update` will be to create a single PR that includes all targets (dependencies) across all folders.
+### No-Consolidation
+To implement a no-consolidation workflow, follow the below pseudo-code
+```
+For each $environment (identified as a glob pattern of folders)
+  run patcher report --include-dirs=$environment
+  for each $target in the output of patcher report
+    run patcher update --target=$target
+```
+
+### Dependency-Only Consolidation
+To implement a dependency-only consolidation workflow, run patcher `update` without any `--include-dirs` arguments.  This will create a single report output JSON file for all updates. Iterate over the targets in that JSON file and run patcher `update` once per identified target.
+
+### Environment-Only Consolidation
+To implement a environment-only consolidation workflow, run patcher `update` for each environment....
+
+### (Environment x Dependency) Consolidation
```

Comment:
no wrong answers - is this intended to be an empty section for now?

Reply:
Still WIP, more to come ;)